src/medCore/database/medDatabaseExporter.cpp

```diff
@@ -59,7 +73,18 @@ void medDatabaseExporter::internalRun(void)
     }
 
     dtkAbstractDataWriter * dataWriter = medAbstractDataFactory::instance()->writer(d->writer);
-    dataWriter->setData(d->data);
+    if(!d->saveMultipleData)
+    {
+        dataWriter->setData(d->data);
+    }
+    else
+    {
+        medAbstractDataWriter* medDataWriter = dynamic_cast<medAbstractDataWriter*>(dataWriter);
+        if(medDataWriter)
```

Comment:
yes but the cast will not fail, as @fcollot with the try/catch, but the original dataWriter could be perhaps empty, i think